src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1389,6 +1389,16 @@ public static bool IsAtStartOfPattern(this SyntaxTree syntaxTree, SyntaxToken le
                 return true;
             }
 
+            // e is > $$
+            // e is >= $$
+            // e is < $$
+            // e is <= $$
+            if (leftToken.IsKind(SyntaxKind.GreaterThanToken, SyntaxKind.GreaterThanEqualsToken, SyntaxKind.LessThanToken, SyntaxKind.LessThanEqualsToken) &&
+                leftToken.Parent.IsKind(SyntaxKind.RelationalPattern))
```

Comment:
I believe there is an existing `IsParentKind` method for `SyntaxToken`s. So you can just do `leftToken.IsParentKind(SyntaxKind.RelationalPattern)`

Reply:
There is ```IsParentKind``` extension method, but only for ```SyntaxNode```